app/frontend/wallet/blockchain-explorer.ts

```diff
@@ -427,9 +430,36 @@ const blockchainExplorer = (ADALITE_CONFIG) => {
     return response
   }
 
-  function getValidStakepools(): Promise<ValidStakePoolsMapping> {
-    const url = `${ADALITE_CONFIG.ADALITE_BLOCKCHAIN_EXPLORER_URL}/api/v2/stakePools`
-    return request(url)
+  async function getValidStakepools(): Promise<StakepoolDataProvider> {
+    const createValidStakePools = async (): Promise<StakepoolDataProvider> => {
```

Comment:
sorry to bother further with it but if we renamed the type to StakepoolDataProvider (which makes sense given its use) the old naming of the related functions/variables - validStakePools seems to be still misleadingly suggesting at first sight it's a collection so I would rename them to `poolDataProvider` or something like that

Reply:
Yes, sorry, I got distracted with some other stuff yesterday and didn't refactor everything that I wanted. This change should be reflected everywhere now.